### ISSUE TITLE
Stop jumping up on refresh

### ISSUE THREAD

**Issue by sinus-x** [sk22/megalodon#340]
**Is your feature request related to a problem? Please describe.**
When the timeline refreshes, or when you trigger the update manually, new posts are loaded on the top and whole timeline jumps up. This is annoying when you want to read whole feed: you have to scroll down, find the post where you stopped reading and start scrolling up again.

**Describe the solution you'd like**
When the timeline loads new posts, they should be loaded on top, "hidden" above the top of the screen. This feature is available at both Twitter app and at the Husky app: when you trigger an update and scroll down just a tiny bit, the top post stays on top and you reveal newer ones by scrolling up, like they have always been there.

**Describe alternatives you've considered**
Another option could be to add a button which would take me back to the last post that was the latest before the refresh. This button could be hidden after some timeout or long manual scrolling action.

**Additional context**
None, I believe.

**Comment by sinus-x** [comment]
This is partially related to #311.

**Comment by sk22** [comment]
100% agree; tried to find out how to do this but i didn't get it to work :/

**Comment by IvanyaK** [comment]
I agree 100% too. In Tusky, when I open the app, it shows the post I stopped at last time.

**Comment by tinsukE** [comment]
Duplicates on upstream:
- https://github.com/mastodon/mastodon-android/issues/70
- https://github.com/mastodon/mastodon-android/issues/147

**Comment by tinsukE** [comment]
I was looking into this, but it would be much more involved than I initially thought:
https://github.com/mastodon/mastodon-android/issues/147#issuecomment-1703834891
